src/meili/map_matcher.cc

```diff
@@ -281,10 +281,7 @@ MatchResult FindMatchResult(const MapMatcher& mapmatcher,
       break;
     node_id = rbegin->nodeid();
     while (rbegin != rend && !prev_edge.Is_Valid()) {
-      // the very first state will have an initial dummy label with invalid edge/node ids
```

Comment:
This comment is actually true, but the loop doesn't look right. Try:
```c++
  for (; std::next(rbegin) != rend; rbegin++) {

// edgeid is always valid
    assert rbegin->edgeid()->IsValid();
    prev_edge = rbegin->edgeid();
  }
```

Reply:
the edgeid will not be vaild when the route is a trivial node to node route as i mentioned below. its a very annoying case unfortunately :frowning_face: